libmproxy/proxy/root_context.py

```diff
@@ -58,6 +57,16 @@ def _next_layer(self, top_layer):
             six.reraise(ProtocolException, ProtocolException(str(e)), sys.exc_info()[2])
         client_tls = is_tls_record_magic(d)
 
+        # 1A. check for --ignore with SNI host
+        if client_tls:
+            try:
+                client_hello = TlsClientHello.from_client_conn(self.client_conn)
+                if (client_hello and
+                        self.config.check_ignore((client_hello.client_sni, 443))):
+                    return RawTCPLayer(top_layer, logging=False)
+            except TlsProtocolException as e:
+                six.reraise(ProtocolException, ProtocolException(str(e)), sys.exc_info()[2])
```

Comment:
I'm afraid we need to mute that - we had some edge cases which we didn't parse properly before, so I don't want to screw up the whole thing because of that.